zio-http/src/test/scala/zhttp/internal/HttpRunnableSpec.scala

```diff
@@ -114,4 +114,22 @@ abstract class HttpRunnableSpec extends ZIOSpecDefault { self =>
         .map(_.status)
     } yield status
   }
+
+  def headers(
+    method: Method = Method.GET,
+    path: Path,
+    headers: Headers = Headers.empty,
+  ): ZIO[EventLoopGroup with ChannelFactory with DynamicServer, Throwable, Headers] = {
+    for {
+      port    <- DynamicServer.port
+      headers <- Client
+        .request(
+          "http://localhost:%d/%s".format(port, path),
+          method,
+          ssl = ClientSSLOptions.DefaultSSL,
+          headers = headers,
+        )
+        .map(_.headers)
+    } yield headers
+  }
```

Comment:
Do we need a integration test for this?

Reply:
Unit test might be sufficient for CORS specific issues but the integration tests will help also in future to catch issues in http encoder/decoder code, or other netty internal changes related to headers